### PR TITLE
chore: dependancy update 

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
Dependancy update for TariCrypto to v0.18
Dependancy update for TariUtilities to v0.5

Motivation and Context
---
This update is mainly required to enable hardware support going forward. 
This update also forces the following breaking dependency API changes:

-  rand 0.7 -> 0.8
- borsh 0.9 -> 0.10
- Digest 0.9 -> 0.10

Most of the changes relate to the hashing API Digest changing. 

How Has This Been Tested?
---
Unit tests

### CODE REVIEW

File: base_layer/mmr/src/merkle_checkpoint.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@
 use std::{convert::TryFrom, fmt, hash::Hasher};
 
 use croaring::Bitmap;
-use digest::Digest;
+use digest::{Digest};
```
Comment (brianp) [inline base_layer/mmr/src/merkle_checkpoint.rs]: This seems wrong.

File: base_layer/mmr/src/functions.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ use crate::{
     MerkleMountainRange,
     MutableMmr,
 };
-use digest::Digest;
+use digest::{Digest};
```
Comment (brianp) [inline base_layer/mmr/src/functions.rs]: This seems wrong. Did it pass clippy?

File: base_layer/mmr/src/mmr_cache.rs [hunk 3]
```diff
@@ -23,7 +23,7 @@
 use std::ops::Deref;
 
 use croaring::Bitmap;
-use digest::Digest;
+use digest::{Digest};
```
Comment (brianp) [inline base_layer/mmr/src/mmr_cache.rs]: Here too

File: base_layer/mmr/src/mutable_mmr.rs [hunk 4]
```diff
@@ -23,7 +23,7 @@
 use std::convert::TryFrom;
 
 use croaring::Bitmap;
-use digest::Digest;
+use digest::{Digest};
```
Comment (brianp) [inline base_layer/mmr/src/mutable_mmr.rs]: And here

File: base_layer/p2p/examples/gen_node_identity.rs [hunk 5]
```diff
@@ -40,7 +40,7 @@ use tari_comms::{
 use tari_utilities::message_format::MessageFormat;
 
 fn random_address() -> Multiaddr {
-    let port = OsRng.gen_range(9000, std::u16::MAX);
+    let port = OsRng.gen_range(9000..std::u16::MAX);
```
Comment (brianp) [inline base_layer/p2p/examples/gen_node_identity.rs:43]: Should this be inclusive?

File: base_layer/tari_mining_helper_ffi/Cargo.toml [hunk 6]
```diff
@@ -8,19 +8,19 @@ edition = "2018"
 
 [dependencies]
 tari_comms = {  path = "../../comms/core" }
-tari_crypto = { version = "0.17" }
+tari_crypto = { version = "0.18" }
 tari_common = {  path = "../../common" }
 tari_core = {  path = "../core", default-features = false, features = ["transactions"]}
-tari_utilities = "0.4.10"
+tari_utilities = { version = "0.5" }
```
Comment (brianp) [inline base_layer/tari_mining_helper_ffi/Cargo.toml:14]:
```suggestion
tari_utilities = "0.5"
```

File: base_layer/wallet/Cargo.toml [hunk 7]
```diff
@@ -12,14 +12,14 @@ tari_common = { path = "../../common" }
...
-tari_crypto = { version = "0.17" }
+tari_crypto = { version = "0.18" }
 tari_key_manager = {  path = "../key_manager", features = ["key_manager_service"] }
 tari_p2p = {  path = "../p2p", features = ["auto-update"] }
 tari_script = { path = "../../infrastructure/tari_script" }
 tari_service_framework = {  path = "../service_framework" }
 tari_shutdown = {  path = "../../infrastructure/shutdown" }
 tari_common_sqlite = { path = "../../common_sqlite" }
-tari_utilities = "0.4.10"
+tari_utilities = { version = "0.5" }
```
Comment (brianp) [inline base_layer/wallet/Cargo.toml:22]:
```suggestion
tari_utilities = "0.5"
```

File: base_layer/wallet_ffi/Cargo.toml [hunk 8]
```diff
@@ -12,12 +12,12 @@ tari_common = { path="../../common" }
...
 tari_comms = {  path = "../../comms/core", features = ["c_integration"]}
 tari_comms_dht = {  path = "../../comms/dht", default-features = false }
-tari_crypto = { version = "0.17" }
+tari_crypto = { version = "0.18" }
 tari_key_manager = {  path = "../key_manager" }
 tari_p2p = {  path = "../p2p" }
 tari_script = { path = "../../infrastructure/tari_script" }
 tari_shutdown = {  path = "../../infrastructure/shutdown" }
-tari_utilities = { version = "0.4.10"}
+tari_utilities = { version = "0.5" }
```
Comment (brianp) [inline base_layer/wallet_ffi/Cargo.toml:20]:
```suggestion
tari_utilities = "0.5"
```

**Review by brianp — APPROVED** [review]
Looks good, minor nits.

File: base_layer/mmr/src/merkle_checkpoint.rs (hunk 1 above)
Comment (hansieodendaal) [reply base_layer/mmr/src/merkle_checkpoint.rs]: Yeah, format should pick it up

**Pull request merged**